corona-warn-app/cwa-app-android
# Additional debug log censoring (EXPOSUREAPP-4451)

Adds censoring for diary locations, persons and the QRCode GUID.

Because diary data is stored asynchronously, we need to delay logging by a little bit such that the data (what to censor) is available to the censor modules.

To improve "dogfooding" for the test builds, I've changed the code to execute what happens in non-tester builds and then just overwrite the censored line again.

## Testing
* Check censoring
    * Scan a QRCode check GUID censoring
    * Create diary persons, should be censored as `Person#<Number>`
    * Create diary locations, should be censored as `Location#<Number>`
    * Export the diary, any logged export data should also be censored
* Check logging of uncaught exceptions again
